LayoutTests/accessibility/date-input-aria-label-and-value.html

```diff
@@ -0,0 +1,100 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
+<html>
+<head>
+<script src="../resources/accessibility-helper.js"></script>
+<script src="../resources/js-test.js"></script>
+</head>
+<body>
+
+<input id="date-input" type="date" value="2022-02-02" />
+
+<script>
+    var testOutput = "Tests that aria-label and aria-valuenow update correctly for date inputs.\n\n";
+
+    function outputString(message) {
+        testOutput += (`\n${message}\n`);
+    }
+
+    function keyDown(key, repeat = 1, message = '') {
+        if (message) 
+            outputString(message + ` (x${repeat})`);
+
+        for (let i = 0 ; i < repeat ; i++) 
+            eventSender.keyDown(key);
+    }
+
+    if (window.accessibilityController) {
+        window.jsTestIsAsync = true;
+        var axDateFieldWrapper = accessibilityController.accessibleElementById("date-input")
+                .childAtIndex(0)
+                .childAtIndex(0);
+
+        setTimeout(async function() {
+            async function expectChildToHaveARIAValueAttributes(childIndex, description) {
+                let result = await expectAsync(`axDateFieldWrapper.childAtIndex(${childIndex}).intValue`, `${description}`);
+
+                result += await expectAsync(`axDateFieldWrapper.childAtIndex(${childIndex}).valueDescription`, `'AXValueDescription: ${description}'`);
+
+                return result;
+            }
+            
+            async function expectChildToHaveARIALabel(childIndex, labelText) {
+                return expectAsync(`platformValueForW3CName(axDateFieldWrapper.childAtIndex(${childIndex}))`, labelText);
+            }
```

Comment:
I don't think this function needs to be async (and it can use `expect` instead of `expectAsync`). The only things that need to be made async in WebKit accessibility layout tests are anything resulting from a dynamic page change via JS or DOM APIs.

Reply:
ok good to know for next time (about `expect` vs `expectAsync`)